src/test/java/com/juniorstart/juniorstart/service/ProjectServiceTest.java

```diff
@@ -0,0 +1,221 @@
+package com.juniorstart.juniorstart.service;
+
+
+import com.juniorstart.juniorstart.model.*;
+import com.juniorstart.juniorstart.model.audit.UserStatus;
+import com.juniorstart.juniorstart.payload.ProjectRequest;
+import com.juniorstart.juniorstart.payload.UserSummary;
+import com.juniorstart.juniorstart.repository.ProjectLikeRepository;
+import com.juniorstart.juniorstart.repository.ProjectRepository;
+import com.juniorstart.juniorstart.repository.TechnologyRepository;
+import com.juniorstart.juniorstart.repository.UserDao;
+import com.juniorstart.juniorstart.security.UserPrincipal;
+import com.juniorstart.juniorstart.util.MockUtil;
+import com.nimbusds.oauth2.sdk.http.HTTPResponse;
+import jakarta.validation.constraints.AssertTrue;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith(MockitoExtension.class)
+public class ProjectServiceTest {
+
+
+    ProjectRepository projectRepository = Mockito.mock(ProjectRepository.class);
+    UserDao userDao = Mockito.mock(UserDao.class);
+    ProjectLikeRepository projectLikeRepository = Mockito.mock(ProjectLikeRepository.class);
+    TechnologyRepository technologyRepository = Mockito.mock(TechnologyRepository.class);
+
+    ProjectService projectService = new ProjectService(projectRepository, userDao, projectLikeRepository, technologyRepository);
+
+    Instant createdAt;
+    Project project;
+    Technology technology;
+    ProjectTodo projectTodo;
```

Comment:
Do każdego testu serwisów. 
Po pierwsze @Mock i @InjectMock.
Po 2 Buildery 
Po 3 statyczne metody.
Po 4 verify 
Po 5 jak masz When a potem assert, to daj enter bo się zlewa
Po 6 Nie lepiej wywołać raz metodę a potem ją sprawdzać w assertTrue, a nie za każdym razem przy każdym assert ją wywoływać ? Bez sensu.
Po 7 KAŻDY TEST W JUNIT 5 !!!!!!!!!!!!!!!!!!!!!! a nie 4 !!!!!!!!!!!!!!!!!!!!!!!!!! Tak się umawialiśmy. 
Po 8 Before wygląda tragicznie źle. Zero przerw między logiką.

Reply:
Dzięki, zdawało mi się, że pisałem w junit5, byłbym wdzięczny jakbyś mi jakiś materiał podesłał.

Reply:
import org.junit.Before;
import org.junit.Test;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.BeforeEach;

Upraszczając na maksa, jak masz org.junit to jest JUnit4 a jak masz org.junit.jupiter to masz JUnit5. Dlatego metodę testów masz w junit 4 Before i Test a BeforeAll i BeforeEach w JUnit5. W sumie Before jest typowe dla JUnit4